[Jun Xiang] iP

JunXiang26 opened the pull request:
#Dubey
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Dubey frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is __FREE__!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

JunXiang26 committed:
Level 0 - Rename, Greet, Exit

JunXiang26 committed:
Level 1 - Echo

JunXiang26 committed:
Level 2 - Add, List

JunXiang26 committed:
Level 3 - Mark as Done

JunXiang26 committed:
Level 4 - ToDos, Events, Deadlines

JunXiang26 committed:
A-TextUiTesting

JunXiang26 committed:
Level 5 - Handle Errors

JunXiang26 committed:
Level 6: Delete

JunXiang26 committed:
Add save feature

JunXiang26 committed:
Merge branch 'branch-Level-7'
Add save feature

JunXiang26 committed:
Update more OOP

JunXiang26 committed:
add dubey package

JunXiang26 committed:
Update README.md, EXPECTED.TXT, input.txt

JunXiang26 committed:
Add . to add-gradle-support

JunXiang26 committed:
Merge branch 'add-gradle-support'

JunXiang26 committed:
Set up Gradle

JunXiang26 committed:
Merge branch 'A-Gradle'

JunXiang26 committed:
Add JUnit Tests

JunXiang26 committed:
Merge branch 'A-JUnit'

JunXiang26 committed:
Add A-Jar

JunXiang26 committed:
Fix no data folder on fresh start bug

JunXiang26 committed:
Add JavaDoc Comments

JunXiang26 committed:
Resolve merge conflict

JunXiang26 committed:
Update A-CodingStandard

JunXiang26 committed:
Resolve A-CodingStandard Conflicts

JunXiang26 committed:
Add find feature

JunXiang26 committed:
Add find feature

jiancheng37 commented on src/main/java/dubey/Dubey.java:
```diff
@@ -0,0 +1,533 @@
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
```
Great job on separating the imports in a way that makes sense!

JunXiang26 replied:
Thanks!

jiancheng37 commented on src/main/java/dubey/Dubey.java:
```diff
@@ -0,0 +1,533 @@
+                    String description = parts[2];
+
+                    switch (type) {
+                    case "T":
```
Great job aligning `case`'s indentation with `switch`

jiancheng37 reviewed (commented):
Overall an excellent job!

JunXiang26 committed:
Separate classes into multiple java files

JunXiang26 committed:
Merge branch 'backup'

JunXiang26 committed:
Set up GUI for Dubey

JunXiang26 committed:
Merge branch 'branch-Level-10'

Inquisitivrx commented on src/main/java/dubey/Dubey.java:
```diff
@@ -0,0 +1,88 @@
+            return ui.showTaskFind(taskList.findAll(keyword));
+        default:
+            throw new IllegalArgumentException("Unknown command: " + command);
+        }
```
Nice work on the neat indentation of the switch cases!

Inquisitivrx commented on src/main/java/dubey/Dubey.java:
```diff
@@ -0,0 +1,88 @@
+     * @param input The user input command.
+     * @return formatted respective String output from processCoammand()
+     */
+    public String getResponse(String input) {
+        try {
+            String response = processCommand(input); // Process command
+            storage.save(taskList.getTasks());
+            return "Processed: " + input + "\n" + response; // Modify this to return a meaningful response
+        } catch (Exception e) {
+            return "Error: " + e.getMessage();
+        }
+    }
```
Good use of Egyptian style brackets!

Inquisitivrx commented on src/main/java/dubey/Storage.java:
```diff
@@ -0,0 +1,106 @@
+package dubey;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
```
Good job on listing imported classes explicitly!

Inquisitivrx reviewed (commented):
Overall, the code structure is well-organized!

JunXiang26 committed:
Resolved Checkstyle Issues

Checkstyle is working now

Checkstyle makes code style more consistent

Updated and Fixed checkstyle issues arising from ./gradlew build

Checkstyle issues caused build to not work previously

JunXiang26 committed:
Test PR on local repo master branch

JunXiang26 committed:
Merge pull request #1 from JunXiang26/branch-A-Assertions

Test PR on local repo master branch

JunXiang26 committed:
Add assertions to document important assumptions

Add assertions to multiple functions

There are several functions that will not work properly
without proper arguements

Ensuring those arguements are correct ensures that these
functions can work properly to reduce bugs

This assertions are implicit checks on the current code

JunXiang26 committed:
Merge pull request #2 from JunXiang26/branch-A-Assertions

Add assertions to document important assumptions

JunXiang26 committed:
Merge branch 'master' into branch-A-CodeQuality

JunXiang26 committed:
Update code quality across all java files

Modularise longer methods.

There are some methods that have very long implementations,
especially in some switch cases.

Modularising these implementations into seperate methods
improve code quality.

This usage of the SLAP method makes code quality better and
code readability easier.

JunXiang26 committed:
Merge pull request #3 from JunXiang26/branch-A-CodeQuality

Update Code Quality

JunXiang26 committed:
Add an extension

JunXiang26 committed:
Merge branch 'branch-BCD-Extension'

JunXiang26 committed:
Remove unused import statement

JunXiang26 committed:
Add title to GUI Window

JunXiang26 committed:
Update User Guide and add Product Snapshot

JunXiang26 committed:
Update Image in README
